tests/components/enphase_envoy/test_binary_sensor.py

```diff
@@ -0,0 +1,62 @@
+"""Test Enphase Envoy binary sensors."""
+
+from unittest.mock import AsyncMock, patch
+
+import pytest
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant.components.enphase_envoy import DOMAIN
+from homeassistant.components.enphase_envoy.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+from homeassistant.setup import async_setup_component
+
+from tests.common import ConfigType, MockConfigEntry
+
+
+@pytest.fixture(name="setup_enphase_envoy_binary_sensor")
+async def setup_enphase_envoy_binary_sensor_fixture(
+    hass: HomeAssistant, config: ConfigType, mock_envoy: AsyncMock
+):
+    """Define a fixture to set up Enphase Envoy with binary sensor platform only."""
+    with (
+        patch(
+            "homeassistant.components.enphase_envoy.Envoy",
+            return_value=mock_envoy,
+        ),
+        patch(
+            "homeassistant.components.enphase_envoy.PLATFORMS",
+            [Platform.BINARY_SENSOR],
+        ),
+    ):
+        assert await async_setup_component(hass, DOMAIN, config)
+        await hass.async_block_till_done()
+        yield
```

Comment:
I think it would make sense to make the `Envoy` mock a fixture (I mean it is already, but a fixture that patches the right packages automatically) and to make this whole a function.

A good example to look at is `flexit_bacnet`. They also use snapshot tests, and they have a nice way to setup the integration for specific platforms.

Also, we should not use the config: ConfigType here, since this isn't a YAML integration anymore, but `flexit_bacnet` is a good example to look from

Reply:
Thanks for the review and suggestion. Used the flexit_bacnet example to refactor this one.